components/count-badge/count-badge.js

```diff
@@ -92,52 +110,153 @@ class CountBadge extends RtlMixin(LitElement) {
-			border-radius: 0.6rem;
+		:host(.focus-visible) .d2l-count-badge-wrapper,
+		.d2l-count-badge-wrapper.focus-visible {
+			box-shadow: 0 0 0 2px var(--d2l-color-celestine);
 		}
 
-		:host([type="notification"]) {
-			background-color: var(--d2l-color-carnelian-minus-1);
+		:host(.focus-visible[icon]) .d2l-count-badge-wrapper,
+		:host([icon]) .d2l-count-badge-wrapper.focus-visible {
+			box-shadow: none;
 		}
 
-		:host([type="count"]) {
-			background-color: var(--d2l-color-gypsum);
+		:host(.focus-visible[icon]) d2l-icon,
+		.d2l-count-badge-wrapper.focus-visible d2l-icon {
+			color: var(--d2l-color-celestine);
+		}
+
+		:host([icon]) .d2l-count-badge-number {
+			position: relative;
+		}
+
+		:host([icon]) .d2l-count-badge-wrapper {
+			padding-left: 2px;
+			padding-right: 14px;
+		}
+
+		:host([size="small"][icon^="tier1"]) .d2l-count-badge-wrapper {
+			max-height: 1.7rem;
+		}
+		:host([size="small"][icon^="tier2"]) .d2l-count-badge-wrapper {
+			max-height: 1.8rem;
+		}
+		:host([size="small"][icon^="tier3"]) .d2l-count-badge-wrapper {
+			max-height: 2rem;
+		}
+		:host([size="large"][icon^="tier1"]) .d2l-count-badge-wrapper {
+			max-height: 2.1rem;
+		}
+		:host([size="large"][icon^="tier2"]) .d2l-count-badge-wrapper {
+			max-height: 2.2rem;
+		}
+		:host([size="large"][icon^="tier3"]) .d2l-count-badge-wrapper {
+			max-height: 2.4rem;
+		}
+		:host([size="small"][icon][fixed-icon-size]) .d2l-count-badge-wrapper {
+			max-height: 1.7rem;
+		}
+		:host([size="large"][icon][fixed-icon-size]) .d2l-count-badge-wrapper {
+			max-height: 2.1rem;
+		}
+
+		:host([size="small"][icon^="tier1"]) .d2l-count-badge-number {
+			left: 0.35rem;
+		}
+		:host([size="small"][icon^="tier2"]) .d2l-count-badge-number {
+			left: 0.65rem;
+		}
+		:host([size="small"][icon^="tier3"]) .d2l-count-badge-number {
+			left: 0.95rem;
+		}
+		:host([size="large"][icon^="tier1"]) .d2l-count-badge-number {
+			left: 0.2rem;
+		}
+		:host([size="large"][icon^="tier2"]) .d2l-count-badge-number {
+			left: 0.5rem;
+		}
+		:host([size="large"][icon^="tier3"]) .d2l-count-badge-number {
+			left: 0.8rem;
+		}
+		:host([size="small"][icon][fixed-icon-size]) .d2l-count-badge-number {
+			left: 0.35rem;
+		}
+		:host([size="large"][icon][fixed-icon-size]) .d2l-count-badge-number {
+			left: 0.2rem;
+		}
+
+		:host([size="small"][icon^="tier1"]) d2l-icon {
+			margin-top: -1.8rem;
+		}
+		:host([size="small"][icon^="tier2"]) d2l-icon {
+			margin-top: -1.5rem;
+		}
+		:host([size="small"][icon^="tier3"]) d2l-icon {
+			margin-top: -1.2rem;
+		}
+		:host([size="large"][icon^="tier1"]) d2l-icon {
+			margin-top: -2.1rem;
+		}
+		:host([size="large"][icon^="tier2"]) d2l-icon {
+			margin-top: -1.9rem;
+		}
+		:host([size="large"][icon^="tier3"]) d2l-icon {
+			margin-top: -1.6rem;
+		}
+		:host([size="small"][icon][fixed-icon-size]) d2l-icon {
+			margin-top: -1.8rem;
+		}
+		:host([size="large"][icon][fixed-icon-size]) d2l-icon {
+			margin-top: -2.1rem;
+		}
```

Comment:
That's a lot of positioning CSS! It's hard to figure out what it's all for... but there must be a better way.

Reply:
Yeah, that's due to the 6 combos we need to support: the three icon tiers and the large and small numbers 😞 Each combo needs different margins to get the number to render 0.55rem/0.65 above and to the right of the icon.

Reply:
One thing you could experiment with would be to make the number absolutely positioned when an icon is present and the `.d2l-count-badge-wrapper` `position: relative` -- all the more reason to split it into 2 components though.